src/WebJobs.Extensions.DurableTask/DurableOrchestrationContextBase.cs

```diff
@@ -454,6 +454,22 @@ public virtual Task CreateTimer(DateTime fireAt, CancellationToken cancelToken)
         /// <param name="input">The JSON-serializeable data to re-initialize the instance with.</param>
         public abstract void ContinueAsNew(object input);
 
+        /// <summary>
+        /// Restarts the orchestration by clearing its history.
+        /// </summary>
+        /// <remarks>
+        /// <para>Large orchestration histories can consume a lot of memory and cause delays in
+        /// instance load times. This method can be used to periodically truncate the stored
+        /// history of an orchestration instance.</para>
+        /// </remarks>
+        /// <param name="input">The JSON-serializeable data to re-initialize the instance with.</param>
+        /// <param name="preserveUnprocessEvents">
+        /// If set to <c>true</c>, re-adds any unprocessed external events into the new execution
+        /// history when the orchestration instance restarts. If <c>false</c>, any unprocessed
+        /// external events will be discarded when the orchestration instance restarts.
+        /// </param>
+        public virtual void ContinueAsNew(object input, bool preserveUnprocessEvents) => throw new NotImplementedException();
```

Comment:
Is there a reason we went with a virtual method that throws `NotImplementedException` as opposed to just making the method abstract?